pkg/application/downloader.go

```diff
@@ -43,13 +44,40 @@ func (downloader) Download(url string) ([]byte, error) {
 	return io.ReadAll(resp.Body)
 }
 
-// GetLatestReleaseVersion returns the latest available version from github
-func (downloader) GetLatestReleaseVersion(releaseURL string) (string, error) {
-	// TODO: Question if there is a less error prone (= simpler) way to install latest avalanchego
-	// Maybe the binary package manager should also allow the actual avalanchego binary for download
-	request, err := http.NewRequest("GET", releaseURL, nil)
+func (d downloader) GetAllReleasesForRepo(org, repo string) ([]string, error) {
+	url := fmt.Sprintf("https://api.github.com/repos/%s/%s/releases", org, repo)
+	body, err := d.doAPIRequest(url)
+	if err != nil {
+		return nil, err
+	}
+	defer body.Close()
+
+	jsonBytes, err := io.ReadAll(body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get latest binary version from %s: %w", url, err)
+	}
+
+	var releaseArr []map[string]interface{}
+	if err := json.Unmarshal(jsonBytes, &releaseArr); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal binary json version string: %w", err)
+	}
+
+	releases := make([]string, len(releaseArr))
+	for i, r := range releaseArr {
+		version := r[githubVersionTagName].(string)
+		if !semver.IsValid(version) {
+			return nil, fmt.Errorf("invalid version string: %s", version)
+		}
+		releases[i] = version
+	}
+
+	return releases, nil
+}
+
+func (downloader) doAPIRequest(url string) (io.ReadCloser, error) {
```

Comment:
is this github specific? If so, consider renaming

Reply:
That's exactly the reason to extract this code: it can be reused for any kind of request.
However, I realize that the error texts suggested a specific use case (most probably because I just moved existing code).
For true generality, I am going to rename the error texts.

I also will make the token a param to the function for true reusability.